syspage.c

```diff
@@ -479,42 +535,52 @@ int syspage_addProg(void *start, void *end, const char *imap, const char *dmap,
 	syspage_common.args[syspage_common.argCnt] = '\0';
 
 	/* copy only program name, without (;) args) */
-	low_memcpy(syspage_common.progs[progID].cmdline, name, pos);
+	low_memcpy(prog->cmdline, name, pos);
 
 	while (pos < MAX_CMDLINE_SIZE)
-		syspage_common.progs[progID].cmdline[pos++] = '\0';
+		prog->cmdline[pos++] = '\0';
 
 	syspage_common.progsCnt++;
 
-	return 0;
+	return ERR_NONE;
 }
 
 
 
 /* Setting kernel's data */
 
-void syspage_setKernelText(void *addr, u32 size)
+void syspage_setKernelText(void *addr, size_t size)
 {
-	syspage_common.syspage->kernel.text = addr;
+	syspage_common.syspage->kernel.text = (addr_t)addr;
 	syspage_common.syspage->kernel.textsz = size;
 
-	syspage_addEntries((u32)addr, size);
+	syspage_addEntries((addr_t)addr, size);
 }
 
 
-void syspage_setKernelBss(void *addr, u32 size)
+void syspage_setKernelBss(void *addr, size_t size)
 {
-	syspage_common.syspage->kernel.bss = addr;
+	syspage_common.syspage->kernel.bss = (addr_t)addr;
 	syspage_common.syspage->kernel.bsssz = size;
 
-	syspage_addEntries((u32)addr, size);
+	syspage_addEntries((addr_t)addr, size);
 }
 
 
-void syspage_setKernelData(void *addr, u32 size)
+void syspage_setKernelData(void *addr, size_t size)
 {
-	syspage_common.syspage->kernel.data = addr;
+	syspage_common.syspage->kernel.data = (addr_t)addr;
 	syspage_common.syspage->kernel.datasz = size;
 
-	syspage_addEntries((u32)addr, size);
+	syspage_addEntries((addr_t)addr, size);
 }
+
+
+/* Add specific architecture data */
+
+void syspage_setArchData(const syspage_arch_t *arch)
+{
+	low_memcpy((void *)&syspage_common.arch, (void *)arch, sizeof(syspage_arch_t));
+}
+
+
```

Comment:
Empty lines at end of the file.

Reply:
Hmm, in the first [commit](https://github.com/phoenix-rtos/plo/pull/44/commits/ba3d8c0fc55b49d3d9fdf46593c7382d70c626c5) gtihub's claimed that there isn't new line at the of the file so that I've added another [one.](https://github.com/phoenix-rtos/plo/compare/96e9fda12b03833586fc74dce7b2b3a86c1f02d1..7a0620f30df0741db7a362ebf28de7bd7d17043b)
I think that i can leave it as it is.